src/frontend/index.js

```diff
@@ -111,34 +107,25 @@ function init() {
 			}
 			
 			persistentConnection.on('connected', function (data) {
-				chunkParser.reset();
 				jsonPerLineParser.reset();
 			});
 			persistentConnection.on('data', function (data) {
-				chunkParser.write(data);
+				jsonPerLineParser.write(data);
 			});
 			persistentConnection.on('end', reconnectOnError);
 			persistentConnection.on('error', reconnectOnError);
 			
-			chunkParser.on('data', function (data) {
-				jsonPerLineParser.write(data);
-			});
-			chunkParser.on('end', reconnectOnError);
-			chunkParser.on('error', reconnectOnError);
-			
 			jsonPerLineParser.on('data', function (data) {
-				data = (data && data.value0 && data.value0.data);
-				
-				if (data && data.length > 0) {
+				if (data && typeof data.x === 'number' && typeof data.y === 'number') {
 					// Advance the time that will go in the next request
 					// to the latest data sample time (add 1 to avoid duplicates):
-					queryParams.since = data[data.length-1].x + 1;
+					queryParams.since = data.x + 1;
 					persistentConnection.setQuery(queryParams);
 					
 					// Notify that the data has been received:
 					dispatcher.emit('receive-data', {
 						dataUrl: dataUrl,
-						data: data
+						data: [ data ]
```

Comment:
Do we need to have ```[ data ]``` as an array here? If the only way ```emit()``` is called is from here, we might as well change ```data: [ data ]``` into ```point: data``` to simplify the rendering code and to and avoid plurality.

(Or do you foresee caching to appear some time soon?)